src/Misc/fileTools.ts

```diff
@@ -493,6 +518,42 @@ export class FileTools {
     public static IsFileURL(): boolean {
         return typeof location !== "undefined" && location.protocol === "file:";
     }
+
+    /**
+     * Test if the given uri is a base64 string
+     * @param uri The uri to test
+     * @return True if the uri is a base64 string or false otherwise
+     */
+    public static IsBase64DataUrl(uri: string): boolean {
+        // Check that the length of the string is at least as long as the minimum base64 data url length
+        if (uri.length >= "data:;base64,".length)

```

Comment:
Unfortunately, this isn't cheap in JS.

```suggestion
        if (uri.length >= 13) // "data:;base64,".length
```

Reply:
As much as I do agree with the change suggested, I am not sure it is as expensive as you think:

![image](https://user-images.githubusercontent.com/1381702/122389009-c1d23e00-cf70-11eb-99a8-2ffb8ca53930.png)

Reply:
Makes sense. I'm just saying it's not free like it is in native code.